tests/tensor/test_subtensor.py

```diff
@@ -105,13 +105,45 @@ def test_as_index_literal():
 
 
 class TestGetCanonicalFormSlice:
+    def test_none_constant(self):
+        slc, rev = get_canonical_form_slice(NoneConst, 5)
+        assert isinstance(slc, slice)
+        assert slc.start == 0
+        assert slc.stop == 5
+        assert slc.step == 1
+        assert rev == 1
+
     def test_scalar_constant(self):
         a = as_scalar(0)
         length = lscalar()
         res = get_canonical_form_slice(a, length)
         assert res[0].owner.op == ptb.switch
         assert res[1] == 1
 
+    def test_tensor_constant(self):
+        a = as_tensor(0)
+        length = lscalar()
+        res = get_canonical_form_slice(a, length)
+        assert res[0].owner.op == ptb.switch
+        assert res[1] == 1
+
+    def test_symbolic_slice(self):
+        idx = make_slice(slice(3, 7, 2))
+        assert not isinstance(idx, slice)
+        res = get_canonical_form_slice(idx, 10)
```

Comment:
Nitpick suggestion for these kinds of tests
```suggestion
        canonical_slice, direction = get_canonical_form_slice(idx, 10)
```

Reply:
Too scared of `black`'s line wrapping wrath